.gitlab-ci.yml

```diff
@@ -689,43 +689,49 @@ dogstatsd_arm64_size_test:
     - mkdir clang && tar xf clang.src.tar.xz --strip-components=1 --no-same-owner -C clang
     - mkdir llvm && tar xf llvm.src.tar.xz --strip-components=1 --no-same-owner -C llvm
     - mkdir build && cd build
-    # Note: on ARM, this may fail (process gets killed), indicating an OOM error.
-    # In that case, reduce this job's parallelism (eg. with -DLLVM_PARALLEL_COMPILE_JOBS=2 -DLLVM_PARALLEL_LINK_JOBS=2)
     - |
       cmake -DLLVM_ENABLE_PROJECTS=clang \
       -DLLVM_TARGETS_TO_BUILD="BPF" \
       -DCMAKE_INSTALL_PREFIX=$DATADOG_AGENT_EMBEDDED_PATH \
       -G "Ninja" \
       -DCMAKE_BUILD_TYPE=MinSizeRel \
       -DLLVM_BUILD_TOOLS=OFF \
+      -DLLVM_ENABLE_TERMINFO=OFF \
       -DLLVM_INCLUDE_EXAMPLES=OFF \
       -DLLVM_INCLUDE_TESTS=OFF \
       -DLLVM_INCLUDE_BENCHMARKS=OFF \
       -DLLVM_STATIC_LINK_CXX_STDLIB=ON \
       -DLLVM_ENABLE_BINDINGS=OFF \
-      -DLLVM_PARALLEL_COMPILE_JOBS=4 \
-      -DLLVM_PARALLEL_LINK_JOBS=4 \
+      -DLLVM_PARALLEL_COMPILE_JOBS=$PARALLEL_JOBS \
+      -DLLVM_PARALLEL_LINK_JOBS=$PARALLEL_JOBS \
       ../llvm
     - cmake --build . --target install
     - cd $DATADOG_AGENT_EMBEDDED_PATH
     - rm -rf bin share libexec lib/clang lib/cmake lib/*.so*
-    - tar cvaf /tmp/clang.tar.xz .
-    - $S3_CP_CMD /tmp/clang.tar.xz $S3_PERMANENT_ARTIFACTS_URI/clang-$ARCH-11.0.0.tar.xz
+    - mkdir -p $CI_PROJECT_DIR/.tmp
+    - tar cvaf $CI_PROJECT_DIR/.tmp/clang-$ARCH-11.0.0.tar.xz .
+    - $S3_CP_CMD $CI_PROJECT_DIR/.tmp/clang-$ARCH-11.0.0.tar.xz $S3_PERMANENT_ARTIFACTS_URI/clang-$ARCH-11.0.0.tar.xz
   timeout: 2h 00m
+  artifacts:
+    expire_in: 2 weeks
+    paths:
+      - $CI_PROJECT_DIR/.tmp/clang-$ARCH-11.0.0.tar.xz
 
 build_clang_x64:
   extends: .build_clang_common
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
   tags: ["runner:main", "size:2xlarge"]
   variables:
     ARCH: amd64
+    PARALLEL_JOBS: 4
 
 build_clang_arm64:
   extends: .build_clang_common
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_arm64:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
   tags: ["runner:docker-arm", "platform:arm64"]
   variables:
     ARCH: arm64
+    PARALLEL_JOBS: 2
```

Comment:
It was discovered that arm64 builds are being arbitrarily limited to 3GB of memory, so this change may no longer be necessary once that is fixed.